### ISSUE TITLE
Module can't be imported in PowerShell Core 7.0

### ISSUE THREAD

**Issue by detlefs** [DarqueWarrior/vsteam#306]
I installed VSTeam 6.4.7

Steps to reproduce
------------------

Start PowerShell Core 7.0.0

```powershell
Import-Module VSTeam
```

Expected behavior
-----------------

Import of the module

Actual behavior
---------------

Error is displayed:

```powershell
Import-Module: The format of value 'Module/ (Windows) PowerShell/7.0.0' is invalid.
```

Environment data
----------------

OS

- [ ] macOS
- [x] Windows
- [ ] Linux

Server

- [ ] TFS 2017
- [ ] TFS 2018
- [x] Azure DevOps Server
- [ ] Azure DevOps Service

```powershell
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.0.0
PSEdition                      Core
GitCommitId                    7.0.0
OS                             Microsoft Windows 10.0.17763
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by SebastianSchuetze** [comment]
I tried it for me and I had no problems. I ran the command in the order you can see in the screenshot:

![image](https://user-images.githubusercontent.com/7256168/81490973-b2ba9000-9288-11ea-883d-627f464024ad.png)

As you can see the only thing that is different is the build number for my Windows 10 OS. Do you have a VM or another PC available to try this?

I have even installed an Azure VM with your Windows Version and installed only PS core 7 and Module Version 6.4.7

Here is a log of my shell

![image](https://user-images.githubusercontent.com/7256168/81492920-582b2f00-929c-11ea-9682-813f82553dae.png)

I currently can't reproduce this issue on two different environments.

**Comment by DarqueWarrior** [comment]
I just tested installing from the PowerShell Gallery and everything worked as expected.

**Comment by detlefs** [comment]
Hi guys,

I did some more experiments today. Here's what I found:
First of all, I updated PowerShell to 7.0.1
Then I tried again, importing the module. Still got the error.
Next, I performed manually the statements from the VSTeam.psm1 file:
```powershell
PowerShell 7.0.1
Copyright (c) Microsoft Corporation. All rights reserved.

https://aka.ms/powershell
Type 'help' to get help.

PowerShell: 7.0.1 - 18.05.2020 13:34:45

Loading personal and system profiles took 3900ms.
§ asgnb007 {~} Import-Module VSTeam
Import-Module: The format of value 'Module/ (Windows) PowerShell/7.0.1' is invalid.
§ asgnb007 {~} Import-Module SHiPS
§ asgnb007 {~} . "C:\Program Files\PowerShell\Modules\VSTeam\6.4.7\vsteam.classes.ps1"
§ asgnb007 {~} . "C:\Program Files\PowerShell\Modules\VSTeam\6.4.7\vsteam.functions.ps1"
§ asgnb007 {~} [VSTeamVersions]::ModuleVersion = _getModuleVersion
§ asgnb007 {~} Set-VSTeamAPIVersion -Target $([VSTeamVersions]::Version)
§ asgnb007 {~} Import-Module VSTeam
§ asgnb007 {~}
```
As you can see, after the manual steps, the module imports without error.

I'm confused about this behavior. If it works manually, it should also work when done in the psm1 file...

**Comment by SebastianSchuetze** [comment]
I found the source of the problem. I get the same error if I called

``` 
Set-VSTeamAccount -Profile MyOrg
Set-VSTeamDefaultProject -Project "MyProject"
```

before I tried to import the module. What happens is that the environment var `$env:TEAM_PROJECT` is filled and then Get-VSTeamProject is called. In there under the [_callAPI](https://github.com/DarqueWarrior/vsteam/blob/13ecf4255d273e9fed49b4c4bf1199bab412b0b3/Source/Private/common.ps1#L8) function is called which invokes [_getUserAgent](https://github.com/DarqueWarrior/vsteam/blob/13ecf4255d273e9fed49b4c4bf1199bab412b0b3/Source/Private/common.ps1#L409). 

The returned value is for me 
```cmd
Invoke-RestMethod: F:\repos\razorspoint_github\vsteam\dist\vsteam.functions.ps1:267:15
Line |
 267 |        $resp = Invoke-RestMethod @params
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~
     | The format of value 'Module/ (Windows) PowerShell/7.0.1' is invalid.
```

So basically when I call `Get-VSTeamProject` I get the same result. So the agent header must be wrong for PS7 + I think.

**Comment by SebastianSchuetze** [comment]
Ah now I found it. I think `$([VSTeamVersions]::ModuleVersion)` is returning and empty string or null.

https://github.com/DarqueWarrior/vsteam/blob/13ecf4255d273e9fed49b4c4bf1199bab412b0b3/Source/Private/common.ps1#L415

**Comment by SebastianSchuetze** [comment]
I will try to fix this.

**Comment by SebastianSchuetze** [comment]
I haven't tested it, but it seems like, the Get-VSTeamProject is called before `[VSTeamVersions]::ModuleVersion = _getModuleVersion` is called. That means the agent header never has the version in it.

**Comment by SebastianSchuetze** [comment]
No I found out that the error does not appear when I change psm1 to the following:

```powershell
# Dot sources all the files in Public, Private and Classes
# Then reads the file names in Public and exports them
# as the fucntions for the module.
# To export an alias you have to manually add it to the
# Export-ModuleMember below.
# The module manifest is using wildcard exports for functions
# and alias so you only have to name the files correctly.

# Files are built using a script in the root folder
. "$PSScriptRoot\vsteam.classes.ps1"
. "$PSScriptRoot\vsteam.functions.ps1"

# Set the module version
[VSTeamVersions]::ModuleVersion = _getModuleVersion

# Load the correct version of the environment variable
Set-VSTeamAPIVersion -Target $([VSTeamVersions]::Version)

# Check to see if the user stored the default project in an environment variable
if ($null -ne $env:TEAM_PROJECT) {
   # Make sure the value in the environment variable still exisits.
   if (Get-VSTeamProject | Where-Object ProjectName -eq $env:TEAM_PROJECT) {
      Set-VSTeamDefaultProject -Project $env:TEAM_PROJECT
   }
}
```

@detlefs could you try the same and then importing it from a fresh shell?

**Comment by detlefs** [comment]
I tried with the modified psm1 as per you last comment.
It works now. The module imports and the functions can be used.

**Comment by SebastianSchuetze** [comment]
Ok cool. Then we have the solution. I will make a PR to correct this.

**Comment by DarqueWarrior** [comment]
Thanks for all the research and hard work on this @SebastianSchuetze I will review the PR now.
